File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxi = 0;
        int prev = 0;
        int diff = 0;
        for(int i=1;i<prices.length;i++){
            if(prices[i]<prices[prev]){
                prev = i;
                diff = 0;
            }else{
                diff = prices[i]-prices[prev];
                maxi = Math.max(maxi, diff);
            }
        }
        return maxi;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        maxi = 0
        curr = 0
        for i in range(1, len(prices)):  
          if(prices[i]>prices[curr]):
            maxi = max(maxi, prices[i]-prices[curr])
          else:
            curr = i
        
        return maxi
 
        """
        :type prices: List[int]
        :rtype: int
        """
        